flink-java/src/main/java/org/apache/flink/api/java/DataSet.java

```diff
@@ -618,9 +620,9 @@ public long count() throws Exception {
 	}
 	
 	/**
-	 * Returns a distinct set of a {@link Tuple} {@link DataSet} using all fields of the tuple.
+	 * Returns a distinct set of a {@link DataSet} using all fields of the tuple.
 	 * <p>
-	 * Note: This operator can only be applied to Tuple DataSets.
+	 * If the input is a {@link Tuple} {@link DataSet}, uses all fields of the tuple.
```

Comment:
Can you change this to "If the input is a composite type (Tuple or Pojo type), distinct is performed on all fields and each field must be a key type."